docs/guide/misc/grib_memory.rst

```diff
@@ -0,0 +1,91 @@
+
+.. code-block:: python
+
+    import earthkit.data
+
+    f = open("test6.grib", "rb")
+    ds = earthkit.data.from_source("stream", f, read_all=True)
+
+.. warning::
+
+    Use this option carefully since your data might not fit into memory.
+
+Reading data from disk and partially store it in memory
+===========================================================
+
+When reading :ref:`grib` data from disk as a :ref:`file source <data-sources-file>` it is represented as a fieldlist and loaded lazily. After the (fast) initial scan for field offsets and lengths no actual fields are created and no data is read into memory. When we start using the fieldlist, e.g. by iterating over the fields, accessing data or metadtata etc., the fields will be created on demand and the related GRIB handles will be loaded from disk when needed. Whether this data or part of it stays in memory depends on the following :ref:`settings <settings>`:
+
+- :ref:`store-grib-fields-in-memory <store-grib-fields-in-memory>`
+- :ref:`grib-handle-cache-size <grib-handle-cache-size>`
+- :ref:`use-grib-metadata-cache <use-grib-metadata-cache>`
+
+.. _store-grib-fields-in-memory:
+
+store-grib-fields-in-memory
+++++++++++++++++++++++++++++
+
+When ``store-grib-fields-in-memory`` is ``True`` (this is the default) once a field is created it will be stored in the fieldlist. Otherwise the field will be created on demand and deleted when going out of scope.
+
+The actual memory used by a field depends on whether it stores the GRIB handle of the related GRIB message. This is controlled by the :ref:`grib-handle-cache-size <grib-handle-cache-size>` settings:
+
+ - When ``grib-handle-cache-size > 0`` the field objects themselves are lightweight and only store the GRIB handle cache index, while the actual GRIB handles are stored in the cache, which is attached to the fieldlist.
+ - When ``grib-handle-cache-size == 0`` the behaviour depends on ``store-grib-fields-in-memory``:
+
+    - when ``store-grib-fields-in-memory`` is ``True`` the fields does not own their GRIB handle but for each call to data and metadata access a new GRIB handle is created and released once the access finished. This can be useful when the fields have to be kept in memory but the memory usage should be low.
+    - when ``store-grib-fields-in-memory`` is ``False`` the fields are created on demand and will store their own GRIB handle in memory until they gets deleted (when going out of scope).
+
+
+.. _grib-handle-cache-size:
+
+grib-handle-cache-size
+++++++++++++++++++++++++++++
+
+When ``grib-handle-cache-size`` is set to a positive value (default is 1) an in-memory GRIB handle cache is attached to the fieldlist. The cache size determines how many GRIB handles are stored in memory. This is an LRU cache so when it is full the least recently used GRIB handle is removed and a new GRIB message is loaded from disk and added to the cache.
```

Comment:
Is this specified in bytes, or the number of handles? The wording is not 100% clear

Reply:
Number of handles. I improved the explanation.